pkg/internal/testhelpers/containers.go

```diff
@@ -79,35 +150,90 @@ func GetReadOnlyConnection(t testing.TB, DBName string) *pgxpool.Pool {
 }
 
 func DbSetup(DBName string, superuser SuperuserStatus) (*pgxpool.Pool, error) {
-	db, err := pgx.Connect(context.Background(), PgConnectURL(defaultDB, Superuser))
+	defaultDb, err := pgx.Connect(context.Background(), PgConnectURL(defaultDB, Superuser))
 	if err != nil {
 		return nil, err
 	}
+	err = func() error {
+		_, err = defaultDb.Exec(context.Background(), fmt.Sprintf("DROP DATABASE IF EXISTS %s", DBName))
+		if err != nil {
+			return err
+		}
+
+		if multinode {
+			dropDistributed := "CALL distributed_exec($$ DROP DATABASE IF EXISTS %s $$, transactional => false)"
+			_, err = defaultDb.Exec(context.Background(), fmt.Sprintf(dropDistributed, DBName))
+			if err != nil {
+				return err
+			}
+		}
+
+		_, err = defaultDb.Exec(context.Background(), fmt.Sprintf("CREATE DATABASE %s OWNER %s", DBName, promUser))
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
 
-	_, err = db.Exec(context.Background(), fmt.Sprintf("DROP DATABASE IF EXISTS %s", DBName))
 	if err != nil {
+		_ = defaultDb.Close(context.Background())
 		return nil, err
 	}
 
-	_, err = db.Exec(context.Background(), fmt.Sprintf("CREATE DATABASE %s OWNER %s", DBName, promUser))
+	err = defaultDb.Close(context.Background())
 	if err != nil {
 		return nil, err
 	}
 
-	err = db.Close(context.Background())
+	ourDb, err := pgx.Connect(context.Background(), PgConnectURL(DBName, Superuser))
 	if err != nil {
 		return nil, err
 	}
 
-	//some docker images may have timescaledb installed in a template, which we don't always want
-	dbDrop, err := pgx.Connect(context.Background(), PgConnectURL(DBName, Superuser))
+	err = func() error {
```

Comment:
since I don't see any defers here why is this run inside a closure?

Reply:
line 232: if we didn't have the closure we'd have to remember to add the Close() for every return (I think I'm going to split out this closure into a real function)